sklearn/preprocessing/tests/test_target.py

```diff
@@ -0,0 +1,137 @@
+    # should pass
+    regr.fit(X, y)
+
+
+def test_transform_target_regressor_friedman():
+    X = friedman[0]
+    y = friedman[1]
+    # pass some functions
+    regr = TransformTargetRegressor(regressor=LinearRegression(),
+                                    func=np.log, inverse_func=np.exp)
+    y_pred = regr.fit(X, y).predict(X)
+    y_tran = np.ravel(regr.transformer_.transform(y))
+    assert_array_almost_equal(np.log(y), y_tran)
+    assert_array_almost_equal(y, np.ravel(regr.transformer_.inverse_transform(
+        y_tran.reshape(-1, 1))))
+    assert_equal(y.shape, y_pred.shape)
+    assert_allclose(y_pred, regr.inverse_func(regr.regressor_.predict(X)))
+    lr = LinearRegression().fit(X, regr.func(y))
+    assert_allclose(y_pred, regr.inverse_func(lr.predict(X)))
+    assert_array_equal(regr.regressor_.coef_.ravel(),
+                       lr.coef_.ravel())
+    # StandardScaler support 1d array while MaxAbsScaler support only 2d array
+    for transformer in (StandardScaler(), MaxAbsScaler()):
+        regr = TransformTargetRegressor(regressor=LinearRegression(),
+                                        transformer=transformer)
+        y_pred = regr.fit(X, y).predict(X)
+        assert_equal(y.shape, y_pred.shape)
+        y_tran = np.ravel(regr.transformer_.transform(y.reshape(-1, 1)))
+        if issubclass(StandardScaler, transformer.__class__):
+            y_mean = np.mean(y)
+            y_std = np.std(y)
+            assert_array_almost_equal((y - y_mean) / y_std, y_tran)
+        assert_array_almost_equal(
+            y, regr.transformer_.inverse_transform(
+                y_tran.reshape(-1, 1)).squeeze())
+        assert_equal(y.shape, y_pred.shape)
+        lr = LinearRegression()
+        transformer2 = clone(transformer)
+        lr.fit(X, transformer2.fit_transform(y.reshape(-1, 1)).squeeze())
+        assert_allclose(y_pred,
+                        transformer2.inverse_transform(
+                            lr.predict(X).reshape(-1, 1)).squeeze())
+        assert_array_equal(regr.regressor_.coef_.squeeze(),
+                           lr.coef_.squeeze())
+
+
+def test_transform_target_regressor_multioutput():
+    X = friedman[0]
+    y = friedman[1]
+    y = np.vstack((y, y ** 2 + 1)).T
+    # pass some functions
+    regr = TransformTargetRegressor(regressor=LinearRegression(),
+                                    func=np.log, inverse_func=np.exp)
+    y_pred = regr.fit(X, y).predict(X)
+    y_tran = regr.transformer_.transform(y)
+    assert_array_almost_equal(np.log(y), y_tran)
+    assert_array_almost_equal(y, regr.transformer_.inverse_transform(y_tran))
+    assert_equal(y.shape, y_pred.shape)
+    assert_allclose(y_pred, regr.inverse_func(regr.regressor_.predict(X)))
+    lr = LinearRegression().fit(X, regr.func(y))
+    assert_allclose(y_pred, regr.inverse_func(lr.predict(X)))
+    assert_array_equal(regr.regressor_.coef_.ravel(),
```

Comment:
It might help to refactor this logic out into a `check_` function. It appears a number of times.